flash.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+"""script for easy update esp8266 firmware with esptool"""
+
+import os
+import subprocess
+import re
+import sys
+
+def exit_with_enter():
+    """function for exit with enter"""
+    print("Press Enter to exit.")
+    input()
+    sys.exit()
+
+
+def check_esptool():
+    """function for check esptool on device"""
+    try:
+        output = subprocess.run(
+            ["esptool", "version"],
+            stdout=subprocess.PIPE,
+            text=True,
+            check=True
+        )
+        if re.search(".*esptool.*", output.stdout) is not None:
+            return True
+        return False
+    except FileNotFoundError:
+        return False
+
+
+def get_number(max_input):
+    """get correct number from 0 to max_input"""
+    num = 0
+    while True:
+        try:
+            num = int(input())
+            if max_input > num >= 0:
+                break
+            raise ValueError
+        except(TypeError, ValueError):
+            print("Print a correct number.")
+    return num
+
+
+def do_esptool_command(command, to_console=False):
+    """do esptool command in console with subprocess"""
+    try:
+        if to_console:
+            return subprocess.run(
+                command,
+                text=True,
+                check=True
+            )
+        return subprocess.run(
+            command,
+            stdout=subprocess.PIPE,
+            text=True,
+            check=True
+        )
+    except subprocess.CalledProcessError as error:
+        print("")
+        if error.returncode == 2:
+            print("Error: firmares are lost. Reload this folder from your source.")
+        elif error.returncode == 1:
+            print("Error: connection lost. Reconnect your board and relaunch script.")
+        elif error.returncode == 5:
+            print("Error: connection lost. Reconnect your board and relaunch script.")
```

Comment:
Duplicate code. You can do this:
```suggestion
elif error.returncode == 1 or error.returncode == 5:
     print("Error: connection lost. Reconnect your board and relaunch script.")
```